### PR TITLE
Allows collector-only servers to disable the UI and query apis

### LINKED ISSUE #1591: Make it possible to disable the query api for collector-only servers
We are deploying a zipkin server to a high-security production environment, and wish to disable the query API (ZipkinQueryApiV1) in order to minimise access to details of our system architecture.

### PR DESCRIPTION
**Pull request by codefromthecrypt**
This adds the ability to set `QUERY_ENABLED=false` for reasons including
security policy.

Fixes #1591

### CODE REVIEW

File: zipkin-autoconfigure/ui/src/main/java/zipkin/autoconfigure/ui/ZipkinUiAutoConfiguration.java [hunk 1]
```diff
@@ -59,6 +60,7 @@
  */
 @Configuration
 @EnableConfigurationProperties(ZipkinUiProperties.class)
+@ConditionalOnProperty(value = "zipkin.ui.enabled", havingValue = "true", matchIfMissing = true)
```
Comment (shakuzen) [inline zipkin-autoconfigure/ui/src/main/java/zipkin/autoconfigure/ui/ZipkinUiAutoConfiguration.java]: Nits: `name` might be more clear than `value` (which is an alias for `name`). Maybe it is just me but `value = x, havingValue = y` reads kind of funny. Also, the default of `havingValue` should be sufficient for our purposes. Specifying `"true"` might mean something like `"TRUE"` doesn't work.

File: zipkin-server/README.md [hunk 2]
```diff
@@ -110,6 +110,7 @@ zipkin-server is a drop-in replacement for the [scala query service](https://git
 [yaml configuration](src/main/resources/zipkin-server-shared.yml) binds the following environment variables from zipkin-scala:
 
     * `QUERY_PORT`: Listen port for the http api and web ui; Defaults to 9411
+    * `QUERY_ENABLED`: False disables the query api and UI assets; Defaults to true
```
Comment (shakuzen) [inline zipkin-server/README.md]: If we do remove the `havingValue` parameter on the `@ConditionalOnProperty` sites, perhaps writing `false` (formatted) makes it clearer we're looking for that specifically to disable querying. With the `havingValue` anything but `true` will disable, I believe.

**Review by shakuzen — COMMENTED** [review]
Looks good as is; just mentioned a couple nits.

**Comment by codefromthecrypt** [comment]
polished up thx to @shakuzen for the advice

**Review by shakuzen — APPROVED** [review]

**Comment by ghost** [comment]
Nice guys! That's pretty much how I would have done it :dancer:

**Pull request merged**